gradio/components.py

```diff
@@ -523,7 +528,7 @@ def postprocess(self, y: float | None):
         if y is None:
             return None
         else:
-            return float(y)
+            return int(y) if self.integer else float(y)
```

Comment:
Let's update the docstring and type hint.

Reply:
Done!